rmarkdowns/Week_4.Rmd

```diff
@@ -257,10 +260,261 @@ For your challenges this week, create the necessary code chunks below. You can c
+  osmdata_sf()
+
+# retrieving data of coastline on oahu
+Oahu_coast <- Oahu_bb %>%
+  opq() %>%
+  add_osm_feature(key = "natural", value = "coastline") %>%
+  osmdata_sf()
+
+
+# retrieving data of national park on Oahu
+Oahu_np <- Oahu_bb %>%
+  opq() %>%
+  add_osm_feature(key = "boundary", value = "national_park") %>%
+  osmdata_sf()
+
+# retrieving data of protected areas on Oahu
+Oahu_protected <- Oahu_bb %>%
+  opq() %>%
+  add_osm_feature(key = "boundary", value = "protected_area") %>%
+  osmdata_sf()
+```
+
+```{r}
+
+# I tried to combine all codes together BUT the result is unacceptable (to me)
+
+# UPDATE: After I completed this assignment and click "Knit", I noticed there was error on below code (too many "+" therefore Knit halted.  After I fixed the code, the result is the same between plot vs plot1  LOL)
+
+plot <- ggplot() +
+  geom_sf(data = Oahu_streets$osm_lines, inherit.aes = FALSE, color = "#1b9e77", size = .4, alpha = .8) +
+  geom_sf(data = Oahu_small_streets$osm_lines, inherit.aes = FALSE, color = "#d95f02", size = .2, alpha = .8) +
+  geom_sf(data = Oahu_coast$osm_lines, inherit.aes = FALSE, color = "#7570b3", size = .8, alpha = .5) +
+  geom_sf(data = Oahu_np$osm_polygons, inherit.aes = FALSE, color = "#e7298a", size = .2, alpha = .8) +
+  geom_sf(data = Oahu_protected$osm_polygons, inherit.aes = FALSE, color = "#66a61e", size = .2, alpha = .8) +
+  geom_point(data = combined_df, aes(x = longitude, y = latitude, color = common_name), size = 3) +
+  scale_shape_manual(values = c(21,22)) + 
+  scale_color_manual(values=c("#8dd3c7", "#bebada"))+
+  theme_linedraw() +
+  labs(title = "Medicinal Plants in Hawaii",
+       subtitle = "Mamaki (Piper Methysticum) and Noni (Morinda Citrifolia)") +
+   geom_sf_text(size = 1, data = Oahu_protected$osm_polygons, aes(label = name)) + #here we are adding some labels for our protected areas for context 
+  coord_sf(xlim = c(-158.28058, -157.64863), ylim = c(21.25482, 21.71201), expand = TRUE) + # setting the limits of our map based on the lat/long we got from our OSM bounding box
+  ggtitle("Medicinal Plants ** Mamaki (Piper Methysticum) and Noni (Morinda Citrifolia) ** in Oahu, Hawaii", subtitle = "Based on iNaturalist Data as of October 14, 2024") +
+  theme_linedraw() +
+  labs(
+    color="Common Name", #this changes the title of our legend
+    x = "Longitude",
+    y = "Latitude"
+  ) 
+
+plot
+```
+
+
+
+```{r}
+
+# Try again.... 
+# What I did differently - I started with the main code first geom_point(data = combined_df...... then print plot. Slowly add 1 layer at a time i.e.  geom sf(data = Oahu_streets, etc), print plot, then add more layer, print plot, etc. 
+
+# Result - Much better... :) 
+# Lesson Learned - "Never give up - Never surrender". 
+
+# Summary of Analysis - The majority of Mamaki and Noni medicinal plants are outside of protected areas in Oahu, Hawaii (assuming all the protected area have been added to Open Street Map)
```

Comment:
Great analysis! Another thing to keep in mind: we limited our results to 100 observations, and that was spread across Hawaiʻi. If you were to run the data retrieval again with just Oʻahu, or went through the steps for downloading bigger datasets manually, you might get a different result :)